spec/jobs/regular/discourse_post_event/bump_topic_spec.rb

```diff
@@ -0,0 +1,79 @@
+
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe Jobs::DiscoursePostEventBumpTopic do
+
+  let(:admin_1) { Fabricate(:user, admin: true) }
+  let(:topic_1) { Fabricate(:topic, user: admin_1) }
+
+  before do
+    freeze_time DateTime.parse('2019-11-10 12:00')
+
+    Jobs.run_immediately!
+
+    SiteSetting.calendar_enabled = true
+    SiteSetting.discourse_post_event_enabled = true
+  end
+
+  describe '#execute' do
+    context 'when params are present' do
+      it 'creates an auto-bump topic timer' do
+        freeze_time
+        subject.execute(topic_id: topic_1.id, date: '2019-12-10 5:00')
+
+        timer = TopicTimer.find_by(topic: topic_1)
+        expect(timer.status_type).to eq(TopicTimer.types[:bump])
+        expect(timer.execute_at).to eq_time(Time.zone.local(2019, 12, 10, 5, 0))
+      end
+    end
+
+    context 'when the topic_id param is missing' do
+      it 'ddoes not throw an error if the date param is present' do
+        expect {
+          subject.execute(date: '2019-12-10 5:00')
+        }.not_to raise_error
+      end
+      it 'does not throw an error if the date param is missing' do
+        expect { subject.execute({}) }.not_to raise_error
+      end
+    end
+
+    context 'when the date param is missing' do
+      before do
+        Fabricate(:topic_timer,
+        status_type: TopicTimer.types[:bump],
+        execute_at: 3.days.from_now,
+        topic: topic_1,
+        user: admin_1
+       )
+      end
+
+      it 'deletes the timer if type is bump' do
```

Comment:
I'm not sure I understand this use case... should the topic timer get deleted if the date parameter is missing? 

Is this the use case for when an event is edited and a topic bump has been removed?

Reply:
This one is not very intuitive, but it's similar to how the [`set_or_create_timer`](https://github.com/discourse/discourse/blob/99fa10eee60dac8cadc54455f2c6a5724c217e34/app/models/topic.rb#L1387)  method and the requests from [`edit-topic-timer.js`](https://github.com/discourse/discourse/blob/6969e9da7e980659b0b6569129ce32fad61fd058/app/assets/javascripts/discourse/app/controllers/edit-topic-timer.js#L228) work. If the time or date isn't set, it will delete the timer.

> Is this the use case for when an event is edited and a topic bump has been removed?

Yes, if you remove the auto-bump topic reminder, it will delete the timer that way.